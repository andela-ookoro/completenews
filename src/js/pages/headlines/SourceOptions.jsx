import React from 'react';
import PropTypes from 'prop-types';

// class to display source
const SourceOptions = ({ id, description = '', name, fetchAvailableSort }) => (
  <div>
    <a className={'browser-default'} title={description}  href={'#'} value={id} onClick={fetchAvailableSort}>
      {name }
    </a>
  </div>
);
SourceOptions.propTypes = {
  id: PropTypes.string.isRequired,
  description: PropTypes.string,
  name: PropTypes.string.isRequired,
  fetchAvailableSort: PropTypes.func.isRequired,
};
export default SourceOptions;
