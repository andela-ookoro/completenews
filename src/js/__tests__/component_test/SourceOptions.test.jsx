import React from 'react';
import { mount } from 'enzyme';
import Renderer from 'react-test-renderer';
import SourceOptions from 'SourceOptions';

test('Component render a button', () => {
  const source = {
    id: 'BBC',
    name: 'BBC News',
    description: 'BBC world news',
  };
  const onClick = jest.fn();
  const component = Renderer.create(
    <SourceOptions
      key={source.id} name={source.name} title={source.description}
      id={source.id} fetchAvailableSort={onClick}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Value of render button should be same as props', () => {
  const source = {
    id: 'BBC',
    name: 'BBC News',
    description: 'BBC world news',
  };
  const onClick = jest.fn();
  const component = mount(
    <SourceOptions
      key={source.id} name={source.name} title={source.description}
      id={source.id} fetchAvailableSort={onClick}
    />,
  );
  const link = component.find('a');
  expect(link.text()).toEqual(source.name);
});

test('Link onclick should execute the function in the prop', () => {
  const source = {
    id: 'BBC',
    name: 'BBC News',
    description: 'BBC world news',
  };
  const onClick = jest.fn();
  const component = mount(
    <SourceOptions
      key={source.id} name={source.name} title={source.description}
      id={source.id} fetchAvailableSort={onClick}
    />,
  );
  const link = component.find('a');
  link.simulate('click');
  expect(onClick).toBeCalled();
});
