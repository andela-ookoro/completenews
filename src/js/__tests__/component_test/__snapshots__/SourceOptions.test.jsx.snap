// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render a button 1`] = `
<div>
  <a
    className="browser-default"
    href="#"
    onClick={[Function]}
    title="BBC world news"
    value="BBC"
  >
    BBC News
  </a>
</div>
`;
